First draft of 2.4 upgrade guide (#375)
cakephp/chronos: docs/en/2-4-upgrade-guide.rst

references to ``ChronosInterface`` with either a reference to
``Cake\Chronos\Chronos`` for datetime instances or ``Cake\Chronos\ChronosDate``
for date instances.

Fewer mutation methods
======================

For historical reasons the chronos classes included many redundant methods. For
example ``addYear()`` and ``addYears()``. In 2.4.0, all of the singular methods
e.g. ``addYear()`` are deprecated. Instead use the plural versions of the
methods e.g. ``addYears()``.

Simpler Date class
==================

When date abstractions were introduced they shared an interfaces with DateTime
classes. This resulted in many no-op methods on dates. For example calling
``setTime()`` on a date would have no effect. In 2.4, all time related methods
(including timezones) are deprecated on date instances. If your application
needs to use the time component of a date, you should use ``Chronos`` instead.

Upcoming removals in 3.0
========================

The following changes will arrive in 3.0, and don't have a simple deprecation
path. Unfortunately these changes will result in hard breaks in 3.0.

Carbon aliases removed
----------------------

> When chronos was started Carbon had no active maintainers. We included

Reviewer: ```suggestion
When Chronos was started Carbon had no active maintainers. We included
```

> When chronos was started Carbon had no active maintainers. We included
> compatiblity aliases in chronos to help users migrate from the un-maintained
> carbon library to chronos. Presently, carbon has active maintainers and we no
> longer feel the need to provide shims.

Reviewer: ```suggestion
When Chronos was started Carbon had no active maintainers. We included
compatiblity aliases in Chronos to help users migrate from the unmaintained
Carbon library to Chronos. Presently, Carbon has active maintainers and we no
longer feel the need to provide shims.
```

No longer extending DateTime
----------------------------

> Historically chronos has extended PHP's ``DateTime`` classes. This has proven to
> be problematic especially for date classes. While chronos will not extend
> PHP's datetime classes or implements the ``DateTimeInterface``, if a method does

Reviewer: ```suggestion
Historically Chronos has extended PHP's ``DateTime`` classes. This has proven to
be problematic especially for date classes. While Chronos will not extend
PHP's ``DateTime`` classes or implements the ``DateTimeInterface``, if a method does
```
